app/models/tag_group.rb

```diff
@@ -53,6 +53,11 @@ def self.find_id_by_slug(slug)
     nil
   end
 
+  # Same as Tag#find_by_name
+  def self.find_by_name_insensitive(name)
+    self.find_by("lower(name) = ?", name.downcase)
```

Comment:
To be safe, we should add an index on `lower(name)`, to match the tags table.

(It's fairly unlikely that sites will have thousands of tag groups... so it's probably not a real issue. But no harm in being safe)